core/src/smartcontracts/wasm.rs

```diff
@@ -340,12 +344,13 @@ impl<'wrld> Runtime<'wrld> {
         let account_id = caller.data().account_id.clone();
         if let Some(validator) = &mut caller.data_mut().validator {
```

Comment:
I believe that desctructuring could help with borrow checker and avoid cloning. 

```suggestion
        let State {
            wsv,
            validator,
            ..
        } = caller.data_mut();
```

I have applied suggestion and it seems to compile without errors.

Reply:
Yes, it return reference.